testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/saml/LogoutTest.java

```diff
@@ -195,6 +215,110 @@ public void testFrontchannelLogoutInSameBrowser() {
+     */
+    @Test
+    public void testSoapBackchannelLogout() {
+        try (SamlBackchannelLogoutReceiver backchannelLogoutReceiver = new SamlBackchannelLogoutReceiver(8082, sales2Rep.getClientId())) {
+            adminClient.realm(REALM_NAME)
+                    .clients().get(sales2Rep.getId())
+                    .update(ClientBuilder.edit(sales2Rep)
+                            .frontchannelLogout(false)
+                            .attribute(SamlProtocol.SAML_SINGLE_LOGOUT_SERVICE_URL_SOAP_ATTRIBUTE, backchannelLogoutReceiver.getUrl())
+                            .attribute(SamlConfigAttributes.SAML_SERVER_SIGNATURE, "true") // sign logout requests
+                            .attribute(SamlConfigAttributes.SAML_FORCE_NAME_ID_FORMAT_ATTRIBUTE, "true") // Force NameID to username
+                            .attribute(SamlConfigAttributes.SAML_NAME_ID_FORMAT_ATTRIBUTE, "username") // Force NameID to username
+                            .build());
+
+            SAMLDocumentHolder samlResponse = prepareLogIntoTwoApps()
+                    .logoutRequest(getAuthServerSamlEndpoint(REALM_NAME), SAML_CLIENT_ID_SALES_POST, POST)
+                    .nameId(nameIdRef::get)
+                    .sessionIndex(sessionIndexRef::get)
+                    .build()
+                    .getSamlResponse(POST);
+
+            assertThat(samlResponse.getSamlObject(), isSamlStatusResponse(JBossSAMLURIConstants.STATUS_SUCCESS));
+            assertLogoutEvent(SAML_CLIENT_ID_SALES_POST);
+
+            // check that the logout request sent to the client is compliant and signed
+            assertTrue(backchannelLogoutReceiver.isLogoutRequestReceived());
+            LogoutRequestType logoutRequest = backchannelLogoutReceiver.getLogoutRequest();
+            assertNotNull(backchannelLogoutReceiver.getLogoutRequest().getSignature());
+            // check nameID
+            assertEquals(logoutRequest.getNameID().getValue(), bburkeUser.getUsername());
+        } catch (Exception ex) {
+            fail("unexpected error");
+        }
+    }
+
+    /** Logout triggered with SOAP binding, request is properly signed */
+    @Test
+    public void testSoapBackchannelLogoutFromSamlClient() {
+        try {
+            adminClient.realm(REALM_NAME)
+                    .clients().get(salesSigRep.getId())
+                    .update(ClientBuilder.edit(salesSigRep)
+                            .frontchannelLogout(false)
+                            .attribute(SamlConfigAttributes.SAML_FORCE_NAME_ID_FORMAT_ATTRIBUTE, "true") // Force NameID to username
+                            .attribute(SamlConfigAttributes.SAML_NAME_ID_FORMAT_ATTRIBUTE, "username") // Force NameID to username
+                            .build());
+
+            SAMLDocumentHolder samlLogoutResponse = prepareLogIntoTwoAppsSig()
+                    .clearCookies() // remove cookies, since SOAP calls do not embed cookie normally
```

Comment:
Can you elaborate on why this is needed?

Reply:
This is to avoid that the cookie is used for retrieving the user session when the logout request is received. I want to ensure that the session index contained in the logout request is used for determining the session to disconnect (i.e. that [this code](https://github.com/keycloak/keycloak/blob/022d2864a6fa71bd97c6f4ac86de178f8bdcfbd6/services/src/main/java/org/keycloak/protocol/saml/SamlService.java#L585) is used).